python/ray/data/_internal/datasource/huggingface_datasource.py

```diff
@@ -118,43 +118,48 @@ def list_parquet_urls_from_dataset(
     def estimate_inmemory_data_size(self) -> Optional[int]:
         return self._dataset.dataset_size
 
+    def _read_dataset(self) -> Iterable[Block]:
+        import numpy as np
+        import pandas as pd
+        import pyarrow
+        # Note: we pass `self` here instead of `self._dataset` because
+        # we need to trigger the try-import logic at the top of file
+        # to avoid import error of dataset_modules.
+        for batch in self._dataset.with_format("arrow").iter(
+            batch_size=self._batch_size
+        ):
```

Comment:
Comment seems to be misaligned with the code -- we don't really pass self anywhere

Reply:
Thanks; updated.